src/mlpack/core/optimizers/cmaes/cmaes.hpp

```diff
@@ -0,0 +1,347 @@
+/**
+ * @file cmaes.hpp
+ * @author Kartik Nighania (GSoC 17 mentor Marcus Edel)
+ *
+ * Covariance Matrix Adaptation Evolution Strategy
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_CMAES_CMAES_HPP
+#define MLPACK_CORE_OPTIMIZERS_CMAES_CMAES_HPP
+
+#include <mlpack/core.hpp>
+#include <cassert>
+#include <cmath>
+#include <cstdio>
+#include <cstring>
+#include <fstream>
+#include <limits>
+#include <iomanip>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <armadillo>
+#include <iostream>
+#include <cfloat>
+
+#include "random.hpp"
+
+namespace mlpack {
+namespace optimization {
+
+template<typename funcType>
+class CMAES
+{
+ public:
+CMAES(funcType& function, arma::mat& start, arma::mat& stdDivs);
+
+double Optimize(arma::mat& arr);
+
+size_t getDimension(void){ return N;}
+
+void getInitialStart(double *arr, size_t dimension)
+{
+for (int i=0; i<N; i++) arr[i] = xstart[i];
+}
+
+void getInitialStandardDeviations(double *arr, size_t dimension)
+{
+for (int i=0; i<N; i++) arr[i] = rgInitialStds[i];
+}
+
+void getStandardDeviations(double *arr)
```

Comment:
it was used to get the covariance diag()
so a user can see the changes as it goes
but ill remove it .

Done